model/src/lib.rs

```diff
@@ -52,26 +58,97 @@ pub struct OrderCreation {
 }
 
 impl OrderCreation {
+    pub const ORDER_TYPE_HASH: [u8; 32] =
```

Comment:
Don't see how lazy_static helps but I'm adding a comment here:

>  // This constant and the signature related functions come from
    // https://github.com/gnosis/gp-v2-contracts/blob/main/src/contracts/libraries/GPv2Encoding.sol .

Reply:
As in you can compute this value, it's the EIP-712 type hash for the Order (the `keccak256` expression from the comment above the link)

Reply:
I see. Coming from the idea that signing is explained in some textual spec that we implement that makes more sense. On the other hand I approached this whole signature algorithm from the smart contract side in which case it makes sense to have the constant here directly. Not sure what is better.

Reply:
I'm going to add this and your other comment below to the issue of reusing web3 or ethabi. Then we have one issue to track converting this to a more high level implementation but we can still merge the PR now so that we have this functionality. It's more of a cleanup issue then.